bundles/org.openhab.binding.energidataservice/README.md

```diff
@@ -0,0 +1,472 @@
+logInfo("Highest price", (result.get("HighestPrice") as Number).doubleValue.toString)
+logInfo("Most expensive start", (result.get("MostExpensiveStart") as Instant).toString)
+
+// This is a simpler version taking advantage of the fact that each interval here represents 0.1 kWh of consumed energy.
+// In this example we have to provide the total duration to make sure we fit the latest end. This is because there is no
+// registered consumption in the last phase.
+val ArrayList<Duration> durationPhases = new ArrayList<Duration>()
+durationPhases.add(Duration.ofMinutes(37))
+durationPhases.add(Duration.ofMinutes(8))
+durationPhases.add(Duration.ofMinutes(4))
+durationPhases.add(Duration.ofMinutes(2))
+durationPhases.add(Duration.ofMinutes(4))
+durationPhases.add(Duration.ofMinutes(36))
+durationPhases.add(Duration.ofMinutes(41))
+
+var Map<String, Object> result = actions.calculateCheapestPeriod(now.toInstant(), now.plusHours(24).toInstant(), Duration.ofMinutes(236), durationPhases, 0.1 | kWh)
+```
+
+:::
+
+::: tab JavaScript
+
+```javascript
+var edsActions = actions.get("energidataservice", "energidataservice:service:energidataservice");
+
+// Get prices and convert to JavaScript Map with Instant string representation as keys.
+var priceMap = new Map();
+utils.javaMapToJsMap(edsActions.getPrices()).forEach((value, key) => {
+    priceMap.set(key.toString(), value);
+});
+
+var hourStart = time.Instant.now().truncatedTo(time.ChronoUnit.HOURS);
+console.log("Current total price excl. VAT: " + priceMap.get(hourStart.toString()));
+
+utils.javaMapToJsMap(edsActions.getPrices("SpotPrice,NetTariff")).forEach((value, key) => {
+    priceMap.set(key.toString(), value);
+});
+console.log("Current spot price + net tariff excl. VAT: " + priceMap.get(hourStart.toString()));
+
+var price = edsActions.calculatePrice(time.Instant.now(), time.Instant.now().plusSeconds(3600), Quantity("150 W"));
+console.log("Total price for using 150 W for the next hour: " + price.toString());
+
+var durationPhases = [];
+durationPhases.push(time.Duration.ofMinutes(37));
+durationPhases.push(time.Duration.ofMinutes(8));
+durationPhases.push(time.Duration.ofMinutes(4));
+durationPhases.push(time.Duration.ofMinutes(2));
+durationPhases.push(time.Duration.ofMinutes(4));
+durationPhases.push(time.Duration.ofMinutes(36));
+durationPhases.push(time.Duration.ofMinutes(41));
+durationPhases.push(time.Duration.ofMinutes(104));
+
+var consumptionPhases = [];
+consumptionPhases.push(Quantity("162.162 W"));
+consumptionPhases.push(Quantity("750 W"));
+consumptionPhases.push(Quantity("1500 W"));
+consumptionPhases.push(Quantity("3000 W"));
+consumptionPhases.push(Quantity("1500 W"));
+consumptionPhases.push(Quantity("166.666 W"));
+consumptionPhases.push(Quantity("146.341 W"));
+consumptionPhases.push(Quantity("0 W"));
+
+var result = edsActions.calculateCheapestPeriod(time.Instant.now(), time.Instant.now().plusSeconds(24*60*60), durationPhases, consumptionPhases);
```

Comment:
@florian-h05 - I was actually surprised to see that this example works, for example with an array of js-joda `Duration`'s being automatically translated into `java.util.List<java.time.Duration>`. It also made we wonder if it could be made in the other direction, so that the `java.util.Map<java.time.Instant, java.math.BigDecimal>` would become a JS map having js-joda `Instant` keys?

Reply:
Sorry, now re-reading this: https://github.com/openhab/openhab-addons/pull/14376#discussion_r1197452833

Reply:
GraalVM is really great when it comes to mapping JS types to Java types. Most mappings are built-in, however some are manually specified (e.g. Duration). 

I’ve just reread the JavaDoc for GraalVM, and to me it sounds that target type mappings only apply to the JS -> Java direction, see https://www.graalvm.org/truffle/javadoc/org/graalvm/polyglot/HostAccess.Builder.html#targetTypeMapping-java.lang.Class-java.lang.Class-java.util.function.Predicate-java.util.function.Function-.